### PR TITLE
Change anim player

### PR DESCRIPTION
**Pull request by NetFoley**
stp rage pas

### CODE REVIEW

File: Scenes/Characters/Character.gd [hunk 1]
```diff
@@ -15,8 +15,9 @@ var pathfinder : Pathfinder = null setget set_pathfinder
 signal pathfinder_changed
 
 ## STATS
-onready var weapon_node : Node2D = get_node_or_null("WeaponsPoint/Weapon")
-onready var shield_node : Node2D = get_node_or_null("WeaponsPoint/Shield")
+onready var weapon_node : Node2D = get_node_or_null("WeaponsPoint/WeaponPoint/Weapon")
+onready var shield_node : Node2D = get_node_or_null("WeaponsPoint/ShieldPoint/Shield")
+onready var weapons_node : Node2D = get_node_or_null("WeaponsPoint")
```
Comment (Weact) [inline Scenes/Characters/Character.gd]:
```suggestion
onready var weapons_node : Node2D = get_node_or_null("WeaponsPoint")
onready var weapon_node : Node2D = weapons_node.get_node_or_null("WeaponPoint/Weapon")
onready var shield_node : Node2D = weapons_node.get_node_or_null("ShieldPoint/Shield")

```

Comment (Weact) [reply Scenes/Characters/Character.gd]: It is better to get the weapons_node first then get the different weapons according to weapons_node node : 
_weapons_node.get_node_or_null(...)_

File: Scenes/Characters/ShieldPoint.gd [hunk 2]
```diff
@@ -5,11 +5,17 @@ func get_class() -> String: return ""
 #### ACCESSORS ####
 
 #### BUILT-IN ####
+func _ready() -> void:
+	$WeaponPoint/Weapon.set_anim_player($AnimationPlayer)
```
Comment (Weact) [inline Scenes/Characters/ShieldPoint.gd:9]:
```suggestion
	weapon_node.set_anim_player($AnimationPlayer)
```

File: Scenes/Characters/Character.gd [hunk 3]
```diff
@@ -289,6 +298,18 @@ func flip():
...
+		yield(self, "ready")
+	if !is_instance_valid($WeaponsPoint/ShieldPoint/Shield):
+		yield(self, "ready")
+		
+	$WeaponsPoint/ShieldPoint/Shield/Sprite.set_flip_v(facing_left)
+	
+	if !is_instance_valid($WeaponsPoint/WeaponPoint/Weapon):
+		yield(self, "ready")
+		
+	$WeaponsPoint/WeaponPoint/Weapon/Sprite.set_flip_h(facing_left)
```
Comment (Weact) [inline Scenes/Characters/Character.gd:312]:
```suggestion
	weapon_node.get_node_or_null("Sprite").set_flip_h(facing_left)
```

File: Scenes/Characters/States/CharacterAttack.gd [hunk 4]
```diff
@@ -12,21 +12,22 @@ func get_class() -> String: return "CharacterAttackState"
...
 func enter_state():
-	if is_instance_valid(owner) and "weapon_node" in owner:
-		owner.weapon_node.hitbox.call_deferred("set_disabled", false)
-		owner.weapon_node.get_node_or_null("AnimationPlayer").play("attack")
+	if is_instance_valid(owner):
+		owner.velocity_factor = 0.3
+		owner.rotation_factor = 0.2
+		if "weapons_node" in owner and "weapon_node" in owner:
+			owner.weapons_node.get_node_or_null("AnimationPlayer").play("attack")
+			owner.weapon_node.hitbox.call_deferred("set_disabled", false)
```
Comment (Weact) [inline Scenes/Characters/States/CharacterAttack.gd]:
```suggestion
	if is_instance_valid(owner):
		owner.velocity_factor = 0.3
		owner.rotation_factor = 0.2
		if is_instance_valid(owner.weapons_node) and is_instance_valid(owner.weapon_node):
			owner.weapons_node.get_node_or_null("AnimationPlayer").play("attack")
			owner.weapon_node.hitbox.call_deferred("set_disabled", false)
```

**Review by Weact — CHANGES_REQUESTED** [review]

**Comment by NetFoley** [comment]
VALIDE OMG

**Review by Weact — APPROVED** [review]

**Pull request merged**